ReactApp/src/components/Editor.jsx

```diff
@@ -35,10 +37,21 @@ import { postMessage } from '../misc/Helpers';
 // Current editor (assumes can be only one instance).
 let editor = {};
 
-function Editor() {
+const POST_MOCK = {
+	id: 1,
+	type: 'post',
+};
+
+function Editor({ post = POST_MOCK }) {
 	const [blocks, setBlocks] = useState([]);
 	const [registeredBlocks, setRegisteredBlocks] = useState([]);
 	const [isCodeEditorEnabled, setCodeEditorEnabled] = useState(false);
+	const titleRef = useRef();
+	const { setupEditor } = useDispatch(editorStore);
+
+	useEffect(() => {
+		setupEditor(post, [], {});
+	}, []);
```

Comment:
I researched briefly, but I was unable to determine why the [web editor invokes](https://github.com/WordPress/gutenberg/blob/e44bc77a252e81a549652e6ab480d62cb8184d5e/packages/editor/src/components/provider/index.js#L235) `setupEditor` within a `useLayoutEffect` call. Any ideas as to why that is or if we should do so?

Reply:
Yeah, it wasn't clear to me why `useLayoutEffect` was used, so I ended up using `useEffect`.

Since we might need to adjust this if we expect changes in the post object (I'm not sure if that happens once the editor is opened on the host app), we could evaluate whether to keep it as it is or follow the same approach. What do you think?

Reply:
Relying upon `useEffect` until we identify a need for `useLayoutEffect` makes sense to me.